packages/react-test-renderer/src/ReactTestRenderer.js

```diff
@@ -557,8 +563,60 @@ const ReactTestRendererFiber = {
   /* eslint-enable camelcase */
 
   unstable_setNowImplementation: setNowImplementation,
+
+  act(callback: () => void): Thenable {
+    let result = batchedUpdates(callback);
+    // warnings copied from ReactTestUtils
+    if (__DEV__) {
+      if (result !== undefined) {
+        let addendum;
+        if (typeof result.then === 'function') {
+          addendum =
+            '\n\nIt looks like you wrote act(async () => ...) or returned a Promise. ' +
+            'Do not write async logic inside act(...)\n';
```

Comment:
Missing period at the end of the sentence.

Also maybe "Putting asynchronous logic inside act() is not yet supported".

Reply:
I think it's the other way around. If we *don't* make it return a thenable now, people can write `await act()` without realizing they're awaiting undefined. *Then* making it actually return a Promise *would* be a breaking change.

By warning on `await` now we're effectively enforcing that you don't depend on what it returns. Therefore we can later make it actually return a Promise in some cases.